packages/react-components/src/WithWalletConnector.ts

```diff
@@ -43,14 +53,21 @@ export function ephemeralConnectorType(create: (c: WithWalletConnector, n: Netwo
  * Note that only the connector is permanent. Individual connections may still be disconnected by the application.
  * @param create Factory function for creating new connector instances.
  */
-export function persistentConnectorType(create: (c: WithWalletConnector, n: Network) => Promise<WalletConnector>) {
+export function persistentConnectorType(
+    create: (
+        c: WithWalletConnector,
+        n: Network,
+        namespaceConfig: WalletConnectNamespaceConfig
```

Comment:
WC config should not be passed in here (or referenced in this class at all). What value would you pass if it's not a WC connector?

The correct way to pass it is by binding the value in a closure like (see [sample](https://github.com/Concordium/concordium-dapp-libraries/blob/a045f3b8bc11c7389190c789072f313686b6c639/samples/sign-message/src/config.ts)):
```typescript
persistentConnectorType((delegate, network) =>
    WalletConnectConnector.create(WALLET_CONNECT_OPTS, delegate, network, WALLET_CONNECT_NAMESPACE_CONFIG)
)
```